### PR TITLE
Remove Kafka TLS sidecar for Zookeeper communication

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Enhancement / new feature

### Description

After dropping support for Kafka 2.4, all our supported Kafka versions now natively support TLS for Zookeeper communication. The TLS sidecar is thanks to that not needed anymore and this PR removes it. 

Removing the PR makes it easier to deal with startups / shutdowns, where the Kafka container required the sidecar to be ready etc. It also simplifies the resource management (resources are defined per container) and other things.

For the record: the sidecar still remains in use in Cruise Control and in Entity Operator pods.

### Checklist

- [x] Write tests
- [x] Make sure all tests pass
- [x] Update documentation
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Update CHANGELOG.md

### CODE REVIEW

**Comment by scholzj** [comment]
@tombentley I can add it if you think ... but we haven't done anything like that for the Zookeeper TLS sidecars.

**Comment by scholzj** [comment]
@strimzi-ci run tests profile=upgrade

**Comment by scholzj** [comment]
@tombentley I added the warning there at startup.

**Comment by tombentley** [comment]
Thanks @scholzj

**Comment by scholzj** [comment]
I think we should remove it after few releases.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/ClusterOperatorConfig.java [hunk 1]
```diff
@@ -92,10 +96,23 @@ public ClusterOperatorConfig(Set<String> namespaces, long reconciliationInterval
...
         return fromMap(map, lookup);
     }
 
+    /**
+     * Logs warnings for removed / deprecated environment variables
+     *
+     * @param map   map from which loading configuration parameters
+     */
+    private static void warningsForRemovedEndVars(Map<String, String> map) {
+        if (map.get(STRIMZI_DEFAULT_TLS_SIDECAR_KAFKA_IMAGE) != null)    {
```
Comment (ppatierno) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/ClusterOperatorConfig.java:110]: just `map.containsKey`? or any specific reason for getting the value and see if it's `null`?

Comment (scholzj) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/ClusterOperatorConfig.java]: It is technically the same, but I changed it to `containsKey`.

**Pull request merged**